engine/engine_vars_test.go

```diff
@@ -371,4 +371,44 @@ func TestEngineVarsHandling(t *testing.T) {
 		},
 		diagtest.Asserts{},
 	)
+	renderTest(
+		t, "required vars in document",
+		[]string{`
+			document "example" {
+				vars {
+					now = "2021-01-01"
+				}
+				required_vars = ["now"]
```

Comment:
Interesting, this might be one of the edge cases -- even though documents can be referenced with `ref` keyword, the resolution of variables might not be obvious. I think it's good to have a test for it!

Reply:
Makes sense, I started with document because the issue mentioned it. I'll work on content and section blocks